gdc_ng_models/alembic/versions/38eab5dd7b0d_extend_entity_sets_with_intent_and_ttl.py

```diff
@@ -0,0 +1,50 @@
+"""extend entity_sets with intent and ttl
+
+Revision ID: 38eab5dd7b0d
+Revises: 12dbbcac7a1d
+Create Date: 2024-04-12 15:51:16.241332
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects import postgresql
+
+from gdc_ng_models.models import entity_set
+
+# revision identifiers, used by Alembic.
+revision = "38eab5dd7b0d"  # pragma: allowlist secret
+down_revision = "12dbbcac7a1d"  # pragma: allowlist secret
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    intent_type = postgresql.ENUM(
```

Comment:
Does intent_type need to be defined again, is it not possible to reference the enum created in the entity_set model similar to line 49?

Reply:
There is a catch-22 here. Defining the model is not the same as defining a migration. If the migration relies on the model you miss the time component if the enumeration changes over time while upgrading and downgrading. It will only work on the latest and greatest model values.